### ISSUE TITLE
Error when loading via AMD in the browser

### ISSUE THREAD

**Issue by natlibfi-arlehiko** [josdejong/workerpool#5]
The code doesn't seem to work in browser (Tested in Firefox):

**HTML**:
```html
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/plain; charset=utf-8"></meta>
    <script src="node_modules/requirejs/require.js" data-main="test"></script>
  </head>
  <body>
  </body>
</html>
```
**Javascript**:
```js
define(['node_modules/workerpool/dist/workerpool'], function(workerpool) {

  'use strict';

  var pool = workerpool.pool();
    
  return pool.exec(function(value) {
    return value + 1;
  }, [1]).then(function(results) {
      
    console.log(results);
    pool.clear();

  });
	
});
```

**Error**: `ReferenceError: require is not defined`

**Comment by josdejong** [comment]
Thanks for reporting. Looks like the library works fine in the browser except when using AMD (requirejs)

The error I get in the browser is:

```
Uncaught Error: Script error for "os", needed by: dist/workerpool
Uncaught Error: Script error for "child_process", needed by: dist/workerpool
```

Will fix it.

**Comment by robhawkes** [comment]
I'm experiencing the same issue – hoping to use this within ViziCities but I need to be able to use ES6 modules (with Babel transpiling) and I get the same issue with the following error in the browser:

```
Uncaught ReferenceError: require is not defined
```

**Comment by josdejong** [comment]
@robhawkes thanks for the feedback. You're working on a cool project :). I suppose your build script does support commonjs (node.js) modules? Are you using WebPack?

**Comment by josdejong** [comment]
@robhawkes this problem may have slipped in in the latest version, v1.2.0. Can you verify whether v1.1.0 works fine or has the same issue?

**Comment by robhawkes** [comment]
Interesting! Yes 1.1.0 works in the browser now when I use ES6 modules, though my build process reports errors (which don't seen to have any negative effect aside from being annoying):

```
ERROR in ./~/workerpool/lib/WorkerHandler.js
Module not found: Error: Cannot resolve module 'child_process' in ./[...]/vizicities/node_modules/workerpool/lib
 @ ./~/workerpool/lib/WorkerHandler.js 58:18-42
```

**Comment by josdejong** [comment]
ah, thanks for testing. That gives me an important clue. Will fix this asap.

**Comment by josdejong** [comment]
I found and fixed two underlying issues, and just released v1.2.1.

@natlibfi-arlehiko and @robhawkes can you please let me know whether this solves both your issues?
